[Docs] [APICTL] Add apictl gen command details and deployment repo introduction (#2949)
wso2/docs-apim: en/docs/install-and-setup/setup/api-controller/advanced-topics/configuring-environment-specific-parameters.md

#  Configuring Environment Specific Parameters

When there are multiple environments, to allow easily configuring environment-specific details, apictl supports an additional parameter file named `api_params.yaml` or an additional directory to store these deployment-related artifacts. It is recommended to store the parameter file inside the deployment directory if there are certificates and other details are included; however, it can be stored somewhere else when certificates are not specified via this additional `api_params.yaml` file. 

## Generating the Deployment Directory 

> When there are multiple artifacts are required to add as deployment-related configurations. It is recommended to use a separate directory to store all these configurations. API controller provides the support to generate this deployment-specific directory using, the following CTL commands.

Reviewer: ```suggestion
When there are multiple artifacts which needs to be added as deployment-related configurations, it is recommended to use a separate directory to store all these configurations. API Controller provides the support to generate this deployment-specific directory using the following commands.
```

-   **Command**
        ``` bash
        apictl gen deployment-dir -s <path-to-API-Source-archive> 
        ```
        ``` bash
        apictl gen deployment-dir -s <path-to-API-Source-archive> -d <path-to-the-Deployment-archive>
        ```
        ``` bash
        apictl gen deployment-dir --source <path-to-API-Source-archive> 
        ```
        ``` bash
        apictl gen deployment-dir --source <path-to-API-Source-archive> --destination <path-to-the-Deployment-archive>
        ```

    !!! info
            **Flags:**  
            
            -   Required :  

>                 `--source` or `-s` : Filepath of the source artifact which is needed to use when generating deployment directory.

Reviewer: ```suggestion
                `--source` or `-s` : File path of the source artifact to be used when generating the deployment directory.
```

            -   Optional :  
                `--destination` or `-d` : Path of the directory where the new deployment directory should be generated.    

    !!! example
            ```bash
            apictl gen deployment-dir  -s  /desktop/source/Dev/PizzaShackAPI_1.0.0_r1   
            ```
            ```bash
            apictl gen deployment-dir  -s /desktop/source/Dev/PizzaShackAPI_1.0.0_r1  -d /desktop/deployment/Dev
            ```
            ```bash
            apictl gen deployment-dir  --source  /desktop/source/Dev/PizzaShackAPI_1.0.0_r1   
            ```
            ```bash
            apictl gen deployment-dir  --source /desktop/source/Dev/PizzaShackAPI_1.0.0_r1  --destination /desktop/deployment/Dev
            ```

    !!!note

>             If the `--destination` flag is not provided, the deployment directory will be generated in the working directory.

Reviewer: ```suggestion
            If the `--destination` flag is not provided, the deployment directory will be generated in the working directory by default.
```

    A project folder with the following default structure will be created in the given directory.

    ``` java
    ├── api_params.yaml
    ├── api_meta.yaml
    └── certificates    
    ```
    <table>
        <thead>
            <tr class="header">
                <th>Sub Directory/File</th>
                <th>Description</th>
            </tr>
        </thead>
        <tbody>
            <tr class="odd">
                <td><code>api_params.yaml</code></td>

>                 <td>The specification of the Environment specific configurations.</td>

Reviewer: ```suggestion
                <td>The specification of the environment specific configurations.</td>
```
